packages/wasm/crate/tests/test_witness.rs

```diff
@@ -0,0 +1,300 @@
+            address: Some(AddressProto { inner: "".into(), alt_bech32m: "penumbra1dugkjttfezh4gfkqs77377gnjlvmkkehusx6953udxeescc0qpgk6gqc0jmrsjq8xphzrg938843p0e63z09vt8lzzmef0q330e5njuwh4290n8pemcmx70sasym0lcjkstgzc".to_string() }),
+            value: Some(Value {
+                amount: 1u64.into(),
+                asset_id: decoded_id,
+            }.into()),
+        }],
+        spends: vec![],
+        swaps: vec![],
+        swap_claims: vec![],
+        delegations: vec![],
+        undelegations: vec![],
+        undelegation_claims: vec![],
+        ibc_relay_actions: vec![],
+        ics20_withdrawals: vec![],
+        position_opens: vec![],
+        position_closes: vec![],
+        position_withdraws: vec![],
+        fee_mode: None,
+        dutch_auction_schedule_actions: vec![],
+        dutch_auction_end_actions: vec![],
+        dutch_auction_withdraw_actions: vec![],
+        delegator_votes: vec![],
+    };
+
+    // Viewing key to reveal asset balances and transactions.
+    let full_viewing_key = FullViewingKey::from_str("penumbrafullviewingkey1mnm04x7yx5tyznswlp0sxs8nsxtgxr9p98dp0msuek8fzxuknuzawjpct8zdevcvm3tsph0wvsuw33x2q42e7sf29q904hwerma8xzgrxsgq2").unwrap();
+
+    let transaction_plan = plan_transaction_inner(
+        storage.clone(),
+        planner_request,
+        full_viewing_key.clone(),
+        *STAKING_TOKEN_ASSET_ID,
+    )
+    .await
+    .unwrap();
+
+    // 2. Generate witness.
+
+    // Retrieve SCT from storage.
+    let tx_last_position: StoredPositionProto = mock_db
+        .get(&tables.tree_last_position, "last_position")
+        .await
+        .unwrap()
+        .unwrap();
+    let tx_last_forgotten: Forgotten = mock_db
+        .get(&tables.tree_last_forgotten, "last_forgotten")
+        .await
+        .unwrap()
+        .unwrap();
+    let tx_tree_commitments: StoreCommitment = mock_db
+        .get(&tables.tree_commitments, "tree_commitments")
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Reconstruct SCT struct.
+    let vec_store_commitments: Vec<StoreCommitment> = vec![tx_tree_commitments.clone()];
+    let sct_proto = StoredTreeProto {
+        last_position: Some(tx_last_position.clone()),
+        last_forgotten: Some(tx_last_forgotten),
+        hashes: [].to_vec(),
+        commitments: vec_store_commitments,
+    };
+
+    let mut sct_tree: StoredTree =
+        serde_wasm_bindgen::from_value(serde_wasm_bindgen::to_value(&sct_proto).unwrap())
+            .expect("roundtrip tree conversion");
+
+    // Generate witness data from SCT and specific transaction plan.
+    witness_inner(transaction_plan.clone(), sct_tree.clone()).unwrap();
+
+    // Modify the witness to use the default position. This will cause a failure
+    // because the position must be witnessed in the tree to be valid.
+    //
+    // It's unclear how to test this because this results in a panic deeper in the
+    // call stack that isn't propogated back to the caller?
```

Comment:
tricky

Reply:
Wish we killed all panics 😢  and exclusively returned Result type...